semgrep-core/src/engine/Match_rules.ml

```diff
@@ -711,7 +711,7 @@ let matches_of_xpatterns config equivalences
 (*****************************************************************************)
 
 (* less: use Set instead of list? *)
-let rec (evaluate_formula :
+let rec (compose_patterns :
```

Comment:
I liked `evaluate_formula` more for this function, IMO `compose_patterns` doesn't capture so well what's really going on. This is actually the function that evaluates the formula. My proposal would be to keep this function as `evaluate_formula` and re-name the other one as `evaluate_formula_on_content` (or similar-is). Just an opinion, maybe just give it another thought.